### ISSUE TITLE
Link Checker Improvements

### ISSUE THREAD

**Issue by DerekMelchin** [QuantConnect/Documentation#1807]
#### Expected Behavior

If a link is empty, then the link checker flags it.
Ex:
```
<a href=''><span class='button-name'>Executable Path: Docker</span> setting</a>
```

#### Actual Behavior

The link checker doesn't flag it.

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

**Comment by LouisSzeto** [comment]
fixed via commit cd7a2bc82213defe570036d3d97d5512d0e574dd

**Comment by AlexCatarino** [comment]
Reopen to add a check for /docs without /docs/v2

**Comment by LouisSzeto** [comment]
fixed via commit 225e960f7dd3591939741b015120d09385bd8726
